### PR TITLE
Toby/refactor audits

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlmesh/core/model/definition.py [hunk 1]
```diff
@@ -1548,6 +1481,20 @@ def depends_on_self(self) -> bool:
 Model = t.Union[SqlModel, SeedModel, PythonModel, ExternalModel]
 
 
+class AuditResult(PydanticModel):
```
Comment (izeigerman) [inline sqlmesh/core/model/definition.py:1484]: The fact that this is located here feels counterintuitive.

File: sqlmesh/utils/jinja.py [hunk 2]
```diff
@@ -358,6 +363,8 @@ def trim(
         for package, names in dependencies_by_package.items():
             result = result.merge(self._trim_macros(names, package))
 
+        result._trimmed = True
```
Comment (izeigerman) [inline sqlmesh/utils/jinja.py:366]: This doesn't get serialized, so we won't know if it's trimmed after the deserialization. Is this intentional?

File: sqlmesh/core/snapshot/evaluator.py [hunk 3]
```diff
@@ -422,16 +423,16 @@ def audit(
...
         if audits_with_args:
             logger.info("Auditing snapshot %s", snapshot.snapshot_id)
 
-        for audit, audit_args in snapshot.audits_with_args:
+        for audit, audit_args in audits_with_args:
             results.append(
                 self._audit(
                     audit=audit,
-                    audit_args=audit_args,
+                    audit_args=audit_args.copy(),
```
Comment (izeigerman) [inline sqlmesh/core/snapshot/evaluator.py:435]: Feels like copy should be doe within `audits_with_args`.

File: tests/core/test_model.py [hunk 4]
```diff
@@ -4554,6 +4555,7 @@ def test_variables_jinja():
         variables={"test_var_a": "test_value", "test_var_d": 1, "test_var_unused": 2},
         jinja_macros=jinja_macros,
     )
+    print(model.python_env)
```
Comment (erindru) [inline tests/core/test_model.py:4558]: Debugging?

File: sqlmesh/core/audit/definition.py [hunk 5]
```diff
@@ -256,8 +264,8 @@ def render_query(
             **{**extra_kwargs, **kwargs},
         )
 
-    def _create_query_renderer(self, node: _Node) -> QueryRenderer:
-        model = t.cast(_Model, node)
+    def _create_query_renderer(self, model: t.Optional[_Model] = None) -> QueryRenderer:
+        assert model
```
Comment (erindru) [inline sqlmesh/core/audit/definition.py:268]: Is this for mypy?

File: sqlmesh/utils/jinja.py (hunk 2 above)
Comment (tobymao) [reply sqlmesh/utils/jinja.py:366]: yea, it's really only used during load from jinja disk

**Review by izeigerman — APPROVED** [review]

File: sqlmesh/migrations/v0059_move_audits_to_model.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            "ttl_ms",
+            "unrestorable",
+        ).from_(snapshots_table),
+        quote_identifiers=True,
+    ):
+        parsed_snapshot = json.loads(snapshot)
+
+        audit_definitions = parsed_snapshot.pop("audits", [])
+        node = parsed_snapshot["node"]
+        node.pop("inline_audits", None)
```
[the remark this answers is not in the document]
Comment (tobymao) [reply sqlmesh/migrations/v0059_move_audits_to_model.py:49]: the inline_audits were already stored in snapshot audits through referenced audits.

**Pull request merged**